### DISCUSSION TITLE
Put hover on file after renaming

### DISCUSSION THREAD

**Discussion by Spreadcat** [jarun/nnn#1889] [Q&A] [answered]
Hi there,

I've got a little plugin to rename a hovered file with a plugin. While the renaming works, the change in the filename and the refresh of the directory after the plugin execution updates the file sorting order. As a result the renamed file changes position in the file list and jumps to a somewhere other position. Currently I have to manually search for the renamed file within the file list in order to continue working.

Q: Is there a way to feed some information back from the plugin into `nnn` in order to keep the file hovered over even after the renaming?

**Comment by 7ocb** [comment]
I was thinking about extending language plugin can talk to `nnn` to add "select file" (hover file) action similar to the "cd to path" action, but not got time to actually implement it.

**Reply by Spreadcat** [reply, marked as answer]
I'll take this as a: "no, there is currently no way for this." Thx. ;)

**Comment by N-R-K** [comment]
> Is there a way to feed some information back from the plugin into nnn in order to keep the file hovered over even after the renaming?

There isn't.

> I was thinking about extending language plugin can talk to nnn to add "select file" (hover file) action similar to the "cd to path" action

Also, if someone wants to implement something like it (pull requests welcome AFAIC), keep in mind that rename can change the directory. E.g `file => ../file` would move the file to another dir.
